wp-redis.php

```diff
@@ -22,6 +22,119 @@
 	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+namespace WPRedis;
+
 if ( defined( 'WP_CLI' ) && WP_CLI && ! class_exists( 'WP_Redis_CLI_Command' ) ) {
 	require_once dirname( __FILE__ ) . '/cli.php';
 }
+
+/**
+ * On activation: Check symlink status on plugin activation, disable the plugin if the file exists already
+ * and is not symlinked properly
+ */
+function activation() {
+
+	if ( file_exists( $db = WP_CONTENT_DIR . '/object-cache.php' ) ) {
+
+		$status = symlink_status();
+
+		if ( false === $status ) {
+
+			$error = sprintf(
+				/* translators: %s: path to object-cache.php file */
+				esc_html__( 'The symlink at %s is no longer pointing to the correct location. Please remove the symlink, then reactivate wp-redis plugin.', 'wp-redis' ),
+				'<code>' . esc_html( WP_CONTENT_DIR . '/object-cache.php' ) . '</code>'
+			);
+
+			deactivate_plugins( __FILE__ );
+			wp_die( $error );
+
+		} elseif ( -1 === $status ) {
+
+			$error = sprintf(
+				/* translators: %s: path to object-cache.php file */
+				esc_html__( 'The file at %s is not a symlink. Please remove the file, then reactivate wp-redis plugin again to create the symlink.', 'wp-redis' ),
+				'<code>' . esc_html( WP_CONTENT_DIR . '/object-cache.php' ) . '</code>'
+			);
+
+			deactivate_plugins( __FILE__ );
+			wp_die( $error );
+
+		}
+	} else {
+
+		if ( function_exists( 'symlink' ) ) {
+			@symlink( plugin_dir_path( __FILE__ ) . 'object-cache.php', $db );
+		}
+	}
+}
+
+/**
+ * On deactivation: remove the symlinked file if found and is pointing to the correct file
```

Comment:
@danielbachhuber Deactivating the plugin should mean stopping its functionality, in which case means removing the symlink ( that's the only purpose of the plugin ). That's what a user would expect IMHO.
The symlink is added again on activation, if the user wishes to activate the functionality again.

As a user, what do you expect if you deactivate the plugin ?